native/src/binaryRoutePlanner.cpp

```diff
@@ -919,96 +944,104 @@ void attachConnectedRoads(RoutingContext* ctx, vector<SHARED_PTR<RouteSegmentRes
 	}
 }
 
-void processOneRoadIntersection(RoutingContext* ctx, SEGMENTS_QUEUE& graphSegments, VISITED_MAP& visitedSegments,
-								double distFromStart, double distanceToEnd, SHARED_PTR<RouteSegment>& segment,
-								int segmentPoint, SHARED_PTR<RouteSegment>& next) {
+bool processOneRoadIntersection(RoutingContext* ctx, bool reverseWaySearch, SEGMENTS_QUEUE& graphSegments,
+								VISITED_MAP& visitedSegments, SHARED_PTR<RouteSegment>& segment,
+								SHARED_PTR<RouteSegment>& next, bool isNullGraph) {
 	if (next.get() != NULL) {
-		double obstaclesTime = ctx->config->router->calculateTurnTime(
-			next,
-			// next->getSegmentStart(),
-			next->isPositive() ? next->road->getPointsLength() - 1 : 0, segment, segmentPoint);
-		distFromStart += obstaclesTime;
+		if (!checkMovementAllowed(ctx, reverseWaySearch, next)) {
+			return false;
+		}
+		float obstaclesTime = (float)ctx->config->router->calculateTurnTime(
```

Comment:
Can be lose 1 during cast double to float. So need to be sure that you do not use so small values here

Reply:
Please add comment here: "//Can lose 1 during cast double to float"